rclcpp/test/benchmark/benchmark_node.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <vector>
+
+#include "performance_test_fixture/performance_test_fixture.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+using performance_test_fixture::PerformanceTest;
+
+class NodePerformanceTest : public PerformanceTest
+{
+public:
+  void SetUp(::benchmark::State & state)
+  {
+    rclcpp::init(0, nullptr);
+    performance_test_fixture::PerformanceTest::SetUp(state);
+  }
+
+  void TearDown(::benchmark::State & state)
+  {
+    rclcpp::shutdown();
+    performance_test_fixture::PerformanceTest::TearDown(state);
+  }
+};
+
+BENCHMARK_F(NodePerformanceTest, create_node)(benchmark::State & state)
+{
+  std::vector<std::unique_ptr<rclcpp::Node>> nodes(10000);
```

Comment:
This vector allocation will cause memory allocation to occur. You should reset the heap counters so that the statistics are reset before the loop runs. If the `heap_allocations` statistic isn't an integer, it's usually an indication that each iteration of your timing loop is not performing exactly the same operations, which will result in noisy results.

```suggestion
  std::vector<std::unique_ptr<rclcpp::Node>> nodes(10000);
  reset_heap_counters();
```